src/HDF5.jl

```diff
@@ -535,25 +536,33 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool,
+function h5open(filename::AbstractString, mode::String,
```

Comment:
ignore ~~also I think the docstring below needs to be here~~ I totally missed the second line of this function defining additional arguments.

I'm not 100% sure on the doc string and where it's best to place it